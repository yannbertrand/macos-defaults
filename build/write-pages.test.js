const fs = require('fs')
jest.mock('fs')

const writePages = require('./write-pages')
require('./handlebars-helpers')

const templatesPath = 'templates'
const destinationPath = 'dist'

describe('write-pages', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('one category, one page', () => {
    describe('with text example', () => {
      describe('with special chars key and values', () => {
        beforeEach(() =>
          callWritePages({
            categories: [
              {
                folder: 'category',
                name: 'Category',
                description: 'This is a description of the category. It can contain "quotes" and markdown.\n\n- a list item',
                keys: [
                  {
                    key: 'a page',
                    domain: 'com.apple.category',
                    title: 'Page',
                    description: 'This is a description of the page. It can contain "quotes" and markdown.\n\n- a list item',
                    param: { type: 'bool' },
                    examples: [
                      {
                        value: '~/Desktop',
                        default: true,
                        text: 'output when value is ~/Desktop',
                      },
                      {
                        value: '"EEE d MMM HH:mm:ss"',
                        text: 'output when value is "EEE d MMM HH:mm:ss"',
                      },
                    ],
                    versions: ['Big Sur'],
                  },
                ],
              },
            ],
          })
        )

        it('should write a category/a-page.md file using the page template', () => {
          const pageReadmeContent = readFile(
            `${destinationPath}/category/a-page.md`
          )
          expect(pageReadmeContent).toMatchSnapshot()
        })
      })

      describe('with requirements', () => {
        beforeEach(() =>
          callWritePages({
            categories: [
              {
                folder: 'category',
                name: 'Category',
                description: 'Category description.',
                keys: [
                  {
                    key: 'page',
                    domain: 'com.apple.category',
                    title: 'Page',
                    description: 'Page description.',
                    param: { type: 'string' },
                    requirements: [
                      {
                        folder: 'another-category',
                        key: 'another-key',
                        name: 'com.apple.category2 another-key',
                        value: true,
                      },
                      {
                        folder: 'a-third-category',
                        key: 'a-third-key',
                        name: 'com.apple.category2 a-third-key',
                        value: 0.5,
                      },
                    ],
                    examples: [
                      {
                        value: true,
                        default: true,
                        text: 'output when value is true',
                      },
                      {
                        value: false,
                        text: 'output when value is false',
                      },
                    ],
                    versions: ['Big Sur'],
                    after: 'killall App',
                  },
                ],
              },
            ],
          })
        )

        it('should write a category/page.md file using the page template', () => {
          const pageReadmeContent = readFile(
            `${destinationPath}/category/page.md`
          )
          expect(pageReadmeContent).toMatchSnapshot()
        })
      })

      describe('with an after command', () => {
        beforeEach(() =>
          callWritePages({
            categories: [
              {
                folder: 'category',
                name: 'Category',
                description: 'Category description.',
                keys: [
                  {
                    key: 'page',
                    domain: 'com.apple.category',
                    title: 'Page',
                    description: 'Page description.',
                    param: { type: 'string' },
                    examples: [
                      {
                        value: '~/Desktop',
                        default: true,
                        text: 'output when value is ~/Desktop',
                      },
                      {
                        value: '~/Pictures',
                        text: 'output when value is ~/Pictures',
                      },
                    ],
                    versions: ['Big Sur'],
                    after: 'killall App',
                  },
                ],
              },
            ],
          })
        )

        it('should write a category/page.md file using the page template', () => {
          const pageReadmeContent = readFile(
            `${destinationPath}/category/page.md`
          )
          expect(pageReadmeContent).toMatchSnapshot()
        })
      })

      describe('with a deleteAlert section', () => {
        beforeEach(() =>
          callWritePages({
            categories: [
              {
                folder: 'category',
                name: 'Category',
                description: 'Category description.',
                keys: [
                  {
                    key: 'page',
                    domain: 'com.apple.category',
                    title: 'Page',
                    description: 'Page description.',
                    param: { type: 'bool' },
                    examples: [],
                    deleteAlert: {
                      type: 'danger',
                      message: 'This an alert message. <br> `defaults command`',
                    },
                    versions: ['Big Sur'],
                  },
                ],
              },
            ],
          })
        )

        it('should write a category/page.md file using the page template', () => {
          const pageReadmeContent = readFile(
            `${destinationPath}/category/page.md`
          )
          expect(pageReadmeContent).toMatchSnapshot()
        })
      })
    })

    describe('with text and possible values example', () => {
      beforeEach(() =>
        callWritePages({
          categories: [
            {
              folder: 'category',
              name: 'Category',
              description: 'Category description.',
              keys: [
                {
                  key: 'page',
                  domain: 'com.apple.category',
                  title: 'Page',
                  description: 'Page description.',
                  param: { type: 'string', values: ['start', 'middle', 'end'] },
                  examples: [
                    {
                      value: 'start',
                      text: 'output when value is start',
                    },
                    {
                      value: 'middle',
                      default: true,
                      text: 'output when value is middle',
                    },
                    {
                      value: 'end',
                      text: 'output when value is end',
                    },
                  ],
                  versions: ['Big Sur'],
                },
              ],
            },
          ],
        })
      )

      it('should write a category/page.md file using the page template', () => {
        const pageReadmeContent = readFile(
          `${destinationPath}/category/page.md`
        )
        expect(pageReadmeContent).toMatchSnapshot()
      })
    })

    describe('with image example', () => {
      beforeEach(() =>
        callWritePages({
          categories: [
            {
              folder: 'category',
              name: 'Category',
              description: 'Category description.',
              keys: [
                {
                  key: 'page',
                  domain: 'com.apple.category',
                  title: 'Page',
                  description: 'Page description.',
                  param: { type: 'bool' },
                  examples: [
                    {
                      value: true,
                      default: true,
                      image: {
                        filename: 'true.png',
                        width: 600,
                        height: 400,
                      },
                    },
                    {
                      value: false,
                      image: {
                        filename: 'false.png',
                        width: 400,
                        height: 200,
                      },
                    },
                  ],
                  versions: ['Big Sur'],
                },
              ],
            },
          ],
        })
      )

      it('should write a category/page.md file using the page template', () => {
        const pageReadmeContent = readFile(
          `${destinationPath}/category/page.md`
        )
        expect(pageReadmeContent).toMatchSnapshot()
      })

      it('should copy the true.png example image to a category/category-page-true folder', () => {
        const originImagePath = readFile(
          `${destinationPath}/category/category-page-true.png`
        )
        expect(originImagePath).toMatchInlineSnapshot(
          `"copied:../images/category/page/true.png"`
        )
      })

      it('should copy the false.png example image to a category/category-page-false folder', () => {
        const originImagePath = readFile(
          `${destinationPath}/category/category-page-false.png`
        )
        expect(originImagePath).toMatchInlineSnapshot(
          `"copied:../images/category/page/false.png"`
        )
      })
    })

    describe('with video example', () => {
      beforeEach(() =>
        callWritePages({
          categories: [
            {
              folder: 'category',
              name: 'Category',
              description: 'Category description.',
              keys: [
                {
                  key: 'page',
                  domain: 'com.apple.category',
                  title: 'Page',
                  description: 'Page description.',
                  param: { type: 'float' },
                  examples: [
                    {
                      value: '0',
                      default: true,
                      video: {
                        filename: '0.mp4',
                        width: 750,
                        height: 400,
                      },
                    },
                    {
                      value: '0.5',
                      video: {
                        filename: '0.5.mp4',
                        width: 720,
                        height: 390,
                      },
                    },
                  ],
                  versions: ['Big Sur'],
                },
              ],
            },
          ],
        })
      )

      it('should write a category/page.md file using the page template', () => {
        const pageReadmeContent = readFile(
          `${destinationPath}/category/page.md`
        )
        expect(pageReadmeContent).toMatchSnapshot()
      })

      it('should copy the 0.mp4 example video to a category/category-page-0 folder', () => {
        const originVideoPath = readFile(
          `${destinationPath}/category/category-page-0.mp4`
        )
        expect(originVideoPath).toMatchInlineSnapshot(
          `"copied:../images/category/page/0.mp4"`
        )
      })

      it('should copy the 0.5.mp4 example video to a category/category-page-0.5 folder', () => {
        const originVideoPath = readFile(
          `${destinationPath}/category/category-page-0.5.mp4`
        )
        expect(originVideoPath).toMatchInlineSnapshot(
          `"copied:../images/category/page/0.5.mp4"`
        )
      })
    })
  })

  describe('one category, two pages', () => {
    beforeEach(() =>
      callWritePages({
        categories: [
          {
            folder: 'category',
            name: 'Category',
            description: 'Category description.',
            keys: [
              {
                key: 'page1',
                domain: 'com.apple.category',
                title: 'Page 1',
                description: 'Page 1 description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    default: true,
                    text: 'output when value is true',
                  },
                  {
                    value: false,
                    text: 'output when value is false',
                  },
                ],
                versions: ['Big Sur'],
              },
              {
                key: 'page2',
                domain: 'com.apple.category',
                title: 'Page 2',
                description: 'Page 2 description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    text: 'output when value is true',
                  },
                  {
                    value: false,
                    default: true,
                    text: 'output when value is false',
                  },
                ],
                versions: ['Big Sur'],
              },
            ],
          },
        ],
      })
    )

    it('should write a category/page1.md file using the page template', () => {
      const pageReadmeContent = readFile(`${destinationPath}/category/page1.md`)
      expect(pageReadmeContent).toMatchSnapshot()
    })

    it('should write a category/page2.md file using the page template', () => {
      const pageReadmeContent = readFile(`${destinationPath}/category/page2.md`)
      expect(pageReadmeContent).toMatchSnapshot()
    })
  })

  describe('two categories, one page in each', () => {
    beforeEach(() =>
      callWritePages({
        categories: [
          {
            folder: 'category1',
            name: 'Category 1',
            description: 'Category 1 description.',
            keys: [
              {
                key: 'page',
                domain: 'com.apple.category1',
                title: 'Page',
                description: 'Page description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    default: true,
                    text: 'output when value is true',
                  },
                  {
                    value: false,
                    text: 'output when value is false',
                  },
                ],
                versions: ['Big Sur'],
              },
            ],
          },
          {
            folder: 'category2',
            name: 'Category 2',
            description: 'Category 2 description.',
            keys: [
              {
                key: 'page',
                domain: 'com.apple.category2',
                title: 'Page',
                description: 'Page description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    text: 'output when value is true',
                  },
                  {
                    value: false,
                    default: true,
                    text: 'output when value is false',
                  },
                ],
                versions: ['Big Sur'],
              },
            ],
          },
        ],
      })
    )

    it('should write a category1/page.md file using the page template', () => {
      const pageReadmeContent = readFile(`${destinationPath}/category1/page.md`)
      expect(pageReadmeContent).toMatchSnapshot()
    })

    it('should write a category2/page.md file using the page template', () => {
      const pageReadmeContent = readFile(`${destinationPath}/category2/page.md`)
      expect(pageReadmeContent).toMatchSnapshot()
    })
  })
})

const callWritePages = (defaults) =>
  writePages({ defaults, url: '/' }, templatesPath, destinationPath)
const readFile = (file) => fs.readFakeFileSync(file, 'utf8')
