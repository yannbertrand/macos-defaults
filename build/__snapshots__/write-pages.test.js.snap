// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write-pages one category, one page with image example should write a category/page.md file using the page template 1`] = `
"---
metaTitle: Page | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page
  - name: "description"
    content: "Page description."
  - property: "og:description"
    content: "Page description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

Page description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

## Set to \`true\` (default value)

\`\`\`bash
defaults write com.apple.category "page" -bool "true" 
\`\`\`
<img
  src="./category-page-true.png"
  alt="Example output with value set to true"
  width="600" height="400" style="height: auto"
/>

## Set to \`false\`

\`\`\`bash
defaults write com.apple.category "page" -bool "false" 
\`\`\`
<img
  src="./category-page-false.png"
  alt="Example output with value set to false"
  width="400" height="200" style="height: auto"
/>

## Read current value
\`\`\`bash
defaults read com.apple.category "page"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category "page"
\`\`\`
"
`;

exports[`write-pages one category, one page with text and possible values example should write a category/page.md file using the page template 1`] = `
"---
metaTitle: Page | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page
  - name: "description"
    content: "Page description."
  - property: "og:description"
    content: "Page description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

Page description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: string
  * start
  * middle
  * end

## Set to \`start\`

output when value is start

\`\`\`bash
defaults write com.apple.category "page" -string "start" 
\`\`\`

## Set to \`middle\` (default value)

output when value is middle

\`\`\`bash
defaults write com.apple.category "page" -string "middle" 
\`\`\`

## Set to \`end\`

output when value is end

\`\`\`bash
defaults write com.apple.category "page" -string "end" 
\`\`\`

## Read current value
\`\`\`bash
defaults read com.apple.category "page"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category "page"
\`\`\`
"
`;

exports[`write-pages one category, one page with text example with a deleteAlert section should write a category/page.md file using the page template 1`] = `
"---
metaTitle: Page | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page
  - name: "description"
    content: "Page description."
  - property: "og:description"
    content: "Page description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

Page description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

## Read current value
\`\`\`bash
defaults read com.apple.category "page"
\`\`\`

## Reset to default value
::: danger
This an alert message. <br> \`defaults command\`
:::
\`\`\`bash
defaults delete com.apple.category "page"
\`\`\`
"
`;

exports[`write-pages one category, one page with text example with an after command should write a category/page.md file using the page template 1`] = `
"---
metaTitle: Page | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page
  - name: "description"
    content: "Page description."
  - property: "og:description"
    content: "Page description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

Page description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: string

## Set to \`~/Desktop\` (default value)

output when value is ~/Desktop

\`\`\`bash
defaults write com.apple.category "page" -string "~/Desktop" && killall App
\`\`\`

## Set to \`~/Pictures\`

output when value is ~/Pictures

\`\`\`bash
defaults write com.apple.category "page" -string "~/Pictures" && killall App
\`\`\`

## Read current value
\`\`\`bash
defaults read com.apple.category "page"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category "page" && killall App
\`\`\`
"
`;

exports[`write-pages one category, one page with text example with requirements should write a category/page.md file using the page template 1`] = `
"---
metaTitle: Page | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page
  - name: "description"
    content: "Page description."
  - property: "og:description"
    content: "Page description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

Page description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: string

## Requirements
- [\`com.apple.category2 another-key\`](../../another-category/another-key.html#set-to-true) must be set to \`true\`
- [\`com.apple.category2 a-third-key\`](../../a-third-category/a-third-key.html#set-to-0.5) must be set to \`0.5\`

## Set to \`true\` (default value)

output when value is true

\`\`\`bash
defaults write com.apple.category "page" -string "true" && killall App
\`\`\`

## Set to \`false\`

output when value is false

\`\`\`bash
defaults write com.apple.category "page" -string "false" && killall App
\`\`\`

## Read current value
\`\`\`bash
defaults read com.apple.category "page"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category "page" && killall App
\`\`\`
"
`;

exports[`write-pages one category, one page with text example with special chars key and values should write a category/a-page.md file using the page template 1`] = `
"---
metaTitle: Page | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page
  - name: "description"
    content: "This is a description of the page. It can contain &quot;quotes&quot; and markdown.\\n\\n- a list item"
  - property: "og:description"
    content: "This is a description of the page. It can contain &quot;quotes&quot; and markdown.\\n\\n- a list item"
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

This is a description of the page. It can contain "quotes" and markdown.

- a list item

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

## Set to \`~/Desktop\` (default value)

output when value is ~/Desktop

\`\`\`bash
defaults write com.apple.category "a page" -bool "~/Desktop" 
\`\`\`

## Set to \`"EEE d MMM HH:mm:ss"\`

output when value is "EEE d MMM HH:mm:ss"

\`\`\`bash
defaults write com.apple.category "a page" -bool "\\"EEE d MMM HH:mm:ss\\"" 
\`\`\`

## Read current value
\`\`\`bash
defaults read com.apple.category "a page"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category "a page"
\`\`\`
"
`;

exports[`write-pages one category, one page with video example should write a category/page.md file using the page template 1`] = `
"---
metaTitle: Page | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page
  - name: "description"
    content: "Page description."
  - property: "og:description"
    content: "Page description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

Page description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: float

## Set to \`0\` (default value)

\`\`\`bash
defaults write com.apple.category "page" -float "0" 
\`\`\`
<video autoplay loop muted playsinline width="750" height="400" style="max-width: 100%; height: auto">
  <source src="./category-page-0.mp4" type="video/mp4">
  Example output with value set to 0
</video>

## Set to \`0.5\`

\`\`\`bash
defaults write com.apple.category "page" -float "0.5" 
\`\`\`
<video autoplay loop muted playsinline width="720" height="390" style="max-width: 100%; height: auto">
  <source src="./category-page-0.5.mp4" type="video/mp4">
  Example output with value set to 0.5
</video>

## Read current value
\`\`\`bash
defaults read com.apple.category "page"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category "page"
\`\`\`
"
`;

exports[`write-pages one category, two pages should write a category/page1.md file using the page template 1`] = `
"---
metaTitle: Page 1 | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page 1
  - name: "description"
    content: "Page 1 description."
  - property: "og:description"
    content: "Page 1 description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page 1

Page 1 description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

## Set to \`true\` (default value)

output when value is true

\`\`\`bash
defaults write com.apple.category "page1" -bool "true" 
\`\`\`

## Set to \`false\`

output when value is false

\`\`\`bash
defaults write com.apple.category "page1" -bool "false" 
\`\`\`

## Read current value
\`\`\`bash
defaults read com.apple.category "page1"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category "page1"
\`\`\`
"
`;

exports[`write-pages one category, two pages should write a category/page2.md file using the page template 1`] = `
"---
metaTitle: Page 2 | Category | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category > Page 2
  - name: "description"
    content: "Page 2 description."
  - property: "og:description"
    content: "Page 2 description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page 2

Page 2 description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

## Set to \`true\`

output when value is true

\`\`\`bash
defaults write com.apple.category "page2" -bool "true" 
\`\`\`

## Set to \`false\` (default value)

output when value is false

\`\`\`bash
defaults write com.apple.category "page2" -bool "false" 
\`\`\`

## Read current value
\`\`\`bash
defaults read com.apple.category "page2"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category "page2"
\`\`\`
"
`;

exports[`write-pages two categories, one page in each should write a category1/page.md file using the page template 1`] = `
"---
metaTitle: Page | Category 1 | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category 1 > Page
  - name: "description"
    content: "Page description."
  - property: "og:description"
    content: "Page description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

Page description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

## Set to \`true\` (default value)

output when value is true

\`\`\`bash
defaults write com.apple.category1 "page" -bool "true" 
\`\`\`

## Set to \`false\`

output when value is false

\`\`\`bash
defaults write com.apple.category1 "page" -bool "false" 
\`\`\`

## Read current value
\`\`\`bash
defaults read com.apple.category1 "page"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category1 "page"
\`\`\`
"
`;

exports[`write-pages two categories, one page in each should write a category2/page.md file using the page template 1`] = `
"---
metaTitle: Page | Category 2 | macOS defaults
meta:
  - property: "og:title"
    content: macOS defaults > Category 2 > Page
  - name: "description"
    content: "Page description."
  - property: "og:description"
    content: "Page description."
  - property: "twitter:card"
    content: "summary"
  - property: "twitter:image"
    content: "https://macos-defaults.netlify.app/media-1x1.webp"
  - property: "og:image"
    content: "https://macos-defaults.netlify.app/media-1x1.jpg"
---
# Page

Page description.

<!-- break lists -->

- **Tested on macOS**:
  * Big Sur
- **Parameter type**: bool

## Set to \`true\`

output when value is true

\`\`\`bash
defaults write com.apple.category2 "page" -bool "true" 
\`\`\`

## Set to \`false\` (default value)

output when value is false

\`\`\`bash
defaults write com.apple.category2 "page" -bool "false" 
\`\`\`

## Read current value
\`\`\`bash
defaults read com.apple.category2 "page"
\`\`\`

## Reset to default value
\`\`\`bash
defaults delete com.apple.category2 "page"
\`\`\`
"
`;
